File: app/src/main/java/com/example/ivan/foursquareapp/models/RecyclerAdapterPlaces.java
package com.example.ivan.foursquareapp.models;


import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.ivan.foursquareapp.GlideApp;
import com.example.ivan.foursquareapp.R;

import java.util.ArrayList;
import java.util.List;

public class RecyclerAdapterPlaces extends RecyclerView.Adapter<RecyclerAdapterPlaces.ViewHolder> {

    private List<ItemVenue> mListVenues = new ArrayList<>();
    private ItemClick mListener;

    public RecyclerAdapterPlaces(ItemClick mListener) {
        this.mListener = mListener;
    }

    public void addVenues(List<ItemVenue> list) {
        mListVenues.addAll(list);
        notifyDataSetChanged();
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_place, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.setPlace(mListVenues.get(position));
    }

    @Override
    public int getItemCount() {
        return mListVenues.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private ImageView imageViewLogo;
        private TextView textViewName, textViewIsOpen, textViewPhone;

        public ViewHolder(View itemView) {
            super(itemView);
            imageViewLogo = itemView.findViewById(R.id.iv_logo);
            textViewName = itemView.findViewById(R.id.tv_name);
            textViewIsOpen = itemView.findViewById(R.id.tv_is_open);
            textViewPhone = itemView.findViewById(R.id.tv_telephone);
            textViewPhone.setOnClickListener(v -> mListener.phoneClick(textViewPhone.getText().toString()));
        }

        public void setPlace(ItemVenue itemVenue) {
            GlideApp
                    .with(imageViewLogo.getContext())
                    .load(itemVenue.getVenue().getPhotos().getGroups().get(0).getItems().get(0).getPrefix() + "600x400" + itemVenue.getVenue().getPhotos().getGroups().get(0).getItems().get(0).getSuffix())
                    .centerCrop()
                    .into(imageViewLogo);
            textViewName.setText(itemVenue.getVenue().getName());
            textViewPhone.setText(itemVenue.getVenue().getContact().getPhone());
            if (itemVenue.getVenue().getHours() != null)
                textViewIsOpen.setText(itemVenue.getVenue().getHours().getIsOpen() ? "open" : "close");
        }
    }

    public interface ItemClick{
        void itemClick();
        void phoneClick(String phone);
    }
}

File: app/src/main/java/com/example/ivan/foursquareapp/ui/adapters/RecyclerAdapterPlaces.kt
package com.example.ivan.foursquareapp.ui.adapters


import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.ivan.foursquareapp.utils.GlideApp
import com.example.ivan.foursquareapp.R
import com.example.ivan.foursquareapp.entity.ItemVenue

import java.util.ArrayList
import kotlinx.android.synthetic.main.item_place.view.*

class RecyclerAdapterPlaces(private val mListener: ItemClick) : RecyclerView.Adapter<RecyclerAdapterPlaces.ViewHolder>() {

    private val mListVenues = ArrayList<ItemVenue>()

    fun addVenues(list: List<ItemVenue>) {
        mListVenues.addAll(list)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_place, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.setPlace(mListVenues[position])
    }

    override fun getItemCount(): Int {
        return mListVenues.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        init {
            itemView.setOnClickListener { mListener.itemClick(mListVenues[adapterPosition].venue!!.id!!) }
            itemView.textViewPhone.setOnClickListener { mListener.phoneClick(itemView.textViewPhone.text.toString()) }
        }

        fun setPlace(itemVenue: ItemVenue) {
            with(itemVenue.venue!!.photos!!.groups[0].items[0]) {
                GlideApp
                        .with(itemView.imageViewLogo.context)
                        .load(prefix + "600x400" + suffix)
                        .centerCrop()
                        .into(itemView.imageViewLogo)
            }
            itemView.textViewName.text = itemVenue.venue!!.name
            itemView.textViewPhone.text = itemVenue.venue!!.contact!!.phone
            if (itemVenue.venue!!.hours != null)
                itemView.textViewOpen.text = if (itemVenue.venue!!.hours!!.isOpen) "open" else "close"
        }
    }

    interface ItemClick {
        fun itemClick(venueId: String)
        fun phoneClick(phone: String)
    }
}
